### DISCUSSION TITLE
Laravel Achievements

### DISCUSSION THREAD

**Discussion by tehwave** [HassanZahirnia/laravel-package-ocean#134] [Package Suggestions]
Hi there!

[Laravel Achievements](https://github.com/tehwave/laravel-achievements) is my package to simply and elegantly add model-based achievements to your Laravel application. It provides a clean and easy-to-use API for creating and managing achievements by using a class-based oriented structure.

While other packages exist to provide this functionality, they come with assumptions about intended usage and complex structures. My package does away with this and makes available an foundation to build upon to fit the application's exact needs.

```bash
php artisan make:achievement UsersFirstPost
```

```php
$user = \App\User::find(1);

$user->achieve(new \App\Achievements\UsersFirstPost());
```

Laravel Achievements can be installed on every version up from Laravel 6 / PHP 7.2 and has full test coverage.

It's my hope that you'd consider including this package as it'll help developers not worry about the boring details of setting up this type of functionality for their application.

**Comment by HassanZahirnia** [comment]
Thanks for the suggestion 😊 I've added the package!
